### PR TITLE
Cex withdraw

### PR DESCRIPTION
**Pull request by rosepuppy**
https://github.com/dydxprotocol/v4-web/assets/1062252/0bc18ddf-2946-4c55-a341-9ec1fb182f68

depends on https://github.com/dydxprotocol/v4-abacus/pull/220 and https://github.com/dydxprotocol/v4-localization/pull/273

---

## Views

* `src/views/dialogs/DepositDialog.tsx`
  * Do not close dialog on deposit

* `src/views/dialogs/OnboardingDialog.tsx`
  * Pipe the eventData for DepositForm

* `src/views/dialogs/DepositDialog/DepositDialogContent.tsx`
  * Turns out we were not tracking deposit events on this entry point D:

* `src/views/forms/AccountManagementForms/DepositForm.tsx`
  * Moved onDeposit callback to after notifaction is added, to ensure notification being added.

* `src/views/forms/AccountManagementForms/TokenSelectMenu.tsx`
  * Hide receipt and search if Exchange is selected

* `src/views/forms/AccountManagementForms/WithdrawForm.tsx`
  * Update token and network select menus to support exchanges
  * add tracking events here
  * Update status notification to indicate exchange

* `src/views/forms/AccountManagementForms/WithdrawForm/WithdrawButtonAndReceipt.tsx`
  * Hide exchange rate and slippage fields for exchange option

* `src/views/forms/NobleDeposit.tsx`
  * Updated according to design
  * hide QR unless acknowledged
  * TODO: add header subtitle text

* `src/views/notifications/TransferStatusNotification/index.tsx`
  * assume notification status is completed for exchange withdraws.

## Components

* `src/components/SearchSelectMenu.tsx`
  * remove `--stickyArea1-topHeight` if `withSearch` is false, since you won't want to make room for the search input

* `src/components/TimeoutButton.tsx`
  * Add `onTimeout` callback

## Constants/Types

* `src/constants/analytics.ts`
  * Track chain and asset type for Deposit and Withdraws

## Functions

* `src/lib/addressUtils.ts`
  * Add function to validate cosmos address

## Hooks

* `src/hooks/useLocalNotifications.tsx`
* `src/hooks/useNotificationTypes.tsx`
  * Do not fetch for squid status if this is an exchange notification. Assume it is already completed. Since the notification is triggered after exchange withdraw Transaction is already completed.

## Packages

* Updated `abacus` and `v4-localization`

### CODE REVIEW

File: src/lib/addressUtils.ts [hunk 1]
```diff
@@ -25,3 +25,22 @@ export function convertBech32Address({
...
+    // Decode the address to verify its structure and prefix
+    const { prefix: decodedPrefix } = fromBech32(address);
+
+    // Check if the decoded address has the expected prefix
+    return decodedPrefix === prefix;
+  } catch (error) {
+    // If decoding fails, the address is not valid
+    return false;
+  }
+}
```
Comment (jaredvu) [inline src/lib/addressUtils.ts:46]: good candidate to add to `tests`

File: src/views/forms/AccountManagementForms/WithdrawForm.tsx [hunk 2]
```diff
@@ -49,6 +49,9 @@ import { getNobleChainId } from '@/lib/squid';
 
 import { TokenSelectMenu } from './TokenSelectMenu';
 import { WithdrawButtonAndReceipt } from './WithdrawForm/WithdrawButtonAndReceipt';
+import { validateCosmosAddress } from '@/lib/addressUtils';
+import { track } from '@/lib/analytics';
+import { AnalyticsEvent } from '@/constants/analytics';
```
Comment (jaredvu) [inline src/views/forms/AccountManagementForms/WithdrawForm.tsx:54]: nit: import order

File: src/views/forms/AccountManagementForms/WithdrawForm.tsx [hunk 3]
```diff
@@ -174,19 +178,28 @@ export const WithdrawForm = () => {
             isCctp
           );
           if (txHash) {
+            const nobleChainId = getNobleChainId();
+            const toChainId = Boolean(exchange) ? nobleChainId : chainIdStr || undefined;
```
Comment (jaredvu) [inline src/views/forms/AccountManagementForms/WithdrawForm.tsx:182]: if there is no `toChainId` is it still a valid transfer?

File: src/components/SearchSelectMenu.tsx [hunk 4]
```diff
@@ -14,6 +14,7 @@ import { WithLabel } from '@/components/WithLabel';
 import { layoutMixins } from '@/styles/layoutMixins';
 import { formMixins } from '@/styles/formMixins';
 import breakpoints from '@/styles/breakpoints';
+import { WithSeparators } from './Separator';
```
Comment (jaredvu) [inline src/components/SearchSelectMenu.tsx:17]: remove unused component

File: src/views/forms/NobleDeposit.tsx [hunk 5]
```diff
@@ -82,23 +85,14 @@ Styled.WithReceipt = styled(WithReceipt)`
   --withReceipt-backgroundColor: var(--color-layer-2);
 `;
 
-Styled.QrCodeContainer = styled.div`
-  display: flex;
-  justify-content: center;
+Styled.QrCode = styled(QrCode)<{ blurred: number }>`
```
Comment (jaredvu) [inline src/views/forms/NobleDeposit.tsx:88]: type for blurred should be boolean

**Review by jaredvu — APPROVED** [review]

File: src/views/forms/NobleDeposit.tsx (hunk 5 above)
Comment (rosepuppy) [reply src/views/forms/NobleDeposit.tsx:88]: oh good call

**Pull request merged**